python/tempo/tsdf.py

```diff
@@ -750,63 +750,63 @@ def asofJoin(
+            left_cols = list(set(left_df.columns).difference(set(self.partitionCols)))
+            right_cols = list(
+                set(right_df.columns).difference(set(right_tsdf.partitionCols))
+            )
 
-                w = Window.partitionBy(*partition_cols).orderBy(
-                    right_prefix + right_tsdf.ts_col
-                )
+            left_prefix = (
+                ""
+                if not left_prefix  # use truthiness of None and ""
+                else left_prefix + "_"
+            )
+            right_prefix = (
+                ""
+                if not right_prefix  # use truthiness of None and ""
+                else right_prefix + "_"
+            )
 
-                new_left_ts_col = left_prefix + self.ts_col
-                new_left_cols = [
-                    f.col(c).alias(left_prefix + c) for c in left_cols
-                ] + partition_cols
-                new_right_cols = [
-                    f.col(c).alias(right_prefix + c) for c in right_cols
-                ] + partition_cols
-                quotes_df_w_lag = right_df.select(*new_right_cols).withColumn(
-                    "lead_" + right_tsdf.ts_col,
-                    f.lead(right_prefix + right_tsdf.ts_col).over(w),
-                )
-                left_df = left_df.select(*new_left_cols)
-                res = (
-                    left_df.join(quotes_df_w_lag, partition_cols)
-                    .where(
-                        left_df[new_left_ts_col].between(
-                            f.col(right_prefix + right_tsdf.ts_col),
-                            f.coalesce(
-                                f.col("lead_" + right_tsdf.ts_col),
-                                f.lit("2099-01-01").cast("timestamp"),
-                            ),
-                        )
+            w = Window.partitionBy(*partition_cols).orderBy(
+                right_prefix + right_tsdf.ts_col
+            )
+
+            new_left_ts_col = left_prefix + self.ts_col
+            new_left_cols = [
+                sfn.col(c).alias(left_prefix + c) for c in left_cols
+            ] + partition_cols
+            new_right_cols = [
+                sfn.col(c).alias(right_prefix + c) for c in right_cols
+            ] + partition_cols
+            quotes_df_w_lag = right_df.select(*new_right_cols).withColumn(
+                "lead_" + right_tsdf.ts_col,
+                sfn.lead(right_prefix + right_tsdf.ts_col).over(w),
+            )
+            left_df = left_df.select(*new_left_cols)
+            res = (
+                left_df.join(quotes_df_w_lag, partition_cols)
+                .where(
+                    left_df[new_left_ts_col].between(
+                        sfn.col(right_prefix + right_tsdf.ts_col),
+                        sfn.coalesce(
+                            sfn.col("lead_" + right_tsdf.ts_col),
+                            sfn.lit("2099-01-01").cast("timestamp"),
+                        ),
```

Comment:
More potential that #337 is no longer fixed since this doesn't appear to just be `fn` -> `sfn` 

Let's double check this